tests/test_integrity.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+
+"""Unit tests for metadata integrity.
+
+Run from the root of the repository using ``tox`` with:
+
+.. code-block:: sh
+
+    $ pip install tox
+    $ tox
+"""
+
+import json
+import os
+import unittest
+from io import StringIO
+from pathlib import Path
+from typing import Union
+
+import yaml
+
+HERE = Path(__file__).parent.resolve()
+ROOT = HERE.parent.resolve()
+
+SCHEMA_PATH = ROOT.joinpath("util", "schema", "registry_schema.json")
+SCHEMA = json.loads(SCHEMA_PATH.read_text())
+
+#: These are the valid licenses listed in the registry schema file
+VALID_LICENSE_LABELS = {
+    value
+    for entry in SCHEMA["properties"]["license"]["anyOf"]
+    for value in entry["properties"]["label"]["enum"]
+}
+VALID_LICENSE_URLS = {
+    value
+    for entry in SCHEMA["properties"]["license"]["anyOf"]
+    for value in entry["properties"]["url"]["enum"]
+}
+#: These ontologies have invalid licenses, but they're grandfathered in
+LEGACY_LICENSE_PREFIXES = {
+    "GSSO",
```

Comment:
be good to have a separate yaml for this

Reply:
(this PR is now moved #1684)